### PR TITLE
Add gettypetypeofmatrix test.

### PR DESCRIPTION
**Pull request by richardlford**
Ported the JIT/Directed/gettypetypeof/gettypetypeofmatrix.cs
test from the desktop clr suite to coreclr.

### CODE REVIEW

**Comment by richardlford** [comment]
@pgavlin, PTAL.

File: tests/src/JIT/Directed/gettypetypeof/gettypetypeofmatrix.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+internal class Test
+{
+    private static object s_null = null;
+    private static object s_object = new object();
+    private static object[] s_objectArray = new object[10];
+    private static Foo s_foo = new Foo();
+    private static Foo[] s_fooArray = new Foo[10];
+
+    static public int Main()
```
Comment (pgavlin) [inline tests/src/JIT/Directed/gettypetypeof/gettypetypeofmatrix.cs]: Nit: s/`static public`/`public static/`

File: tests/src/JIT/Directed/gettypetypeof/gettypetypeofmatrix.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    {
+        int iReturn = 100;
```
Comment (pgavlin) [inline tests/src/JIT/Directed/gettypetypeof/gettypetypeofmatrix.cs]: Nit: remove Hungarian notation

**Comment by pgavlin** [comment]
LGTM aside from two nits.

**Comment by richardlford** [comment]
I accidentally merged this in without fixing the nits. I'll fix them now.

**Pull request merged**